client/my-sites/themes/search-themes-tracks.js

```diff
@@ -0,0 +1,109 @@
+import { usePrevious } from '@wordpress/compose';
+import { useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { isRequestingThemesForQuery, prependThemeFilterKeys } from 'calypso/state/themes/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+function useThemeSearchDetails( query ) {
+	const search = query.search;
+	const searchTaxonomies = useSelector( ( state ) =>
+		prependThemeFilterKeys( state, query.filter )
+	);
+	const searchTerm = searchTaxonomies + search;
```

Comment:
There might be a need to cast `search` to string here, when we load the page the `searchTerm` is being logged as `"undefined"`, which is probably caused by doing something like `'' + undefined`.

Another case is when we select a category we end up with:

```
search: undefined
search_taxonomies: "subject:business"
search_term: "subject:business undefined"
```


When it should be: 

```
search: undefined (or '')
search_taxonomies: "subject:business"
search_term: "subject:business"
```

Reply:
Ooh, good catch, I guess this is what's failing the unit tests! 🤔 

I had something to guard against undefined `search` at some point, but then refactored everything and must have totally lost it in the process.